hail/src/main/scala/is/hail/io/index/IndexWriter.scala

```diff
@@ -380,4 +386,38 @@ class StagedIndexWriter(branchingFactor: Int, keyType: PType, annotationType: PT
     internalBuilder.create(cb)
     internalBuilder.store(cb, utils, 0)
   }
+
+  class NonImperativeIndexWriter(newInit: EmitMethodBuilder[_], newAdd: EmitMethodBuilder[_], newClose: EmitMethodBuilder[_]) {
```

Comment:
Why is this necessary? It seems like we can create a code builder and use the other version of the index writer.